apps/server/src/modules/authentication/strategy/ldap.strategy.ts

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ICurrentUser } from '@shared/domain';
+import { CurrentUserMapper } from '@shared/domain/mapper/current-user.mapper';
+import { SchoolRepo, SystemRepo, UserRepo } from '@shared/repo';
+import { Strategy } from 'passport-custom';
+import { AuthenticationService } from '../services/authentication.service';
+import { LdapService } from '../services/ldap.service';
+
+export type RequestBody = { systemId?: string; username?: string; password?: string; schoolId?: string };
+
+@Injectable()
+export class LdapStrategy extends PassportStrategy(Strategy, 'ldap') {
+	constructor(
+		private readonly systemRepo: SystemRepo,
+		private readonly schoolRepo: SchoolRepo,
+		private readonly ldapService: LdapService,
+		private readonly authenticationService: AuthenticationService,
+		private readonly userRepo: UserRepo
+	) {
+		super();
+	}
+
+	async validate(request: { body: RequestBody }): Promise<ICurrentUser> {
+		const { systemId, schoolId } = request.body;
+		let { username, password } = request.body;
+		if (!systemId || !username || !password || !schoolId) {
+			throw new UnauthorizedException();
+		}
+		username = this.authenticationService.normalizeUsername(username);
+		password = this.authenticationService.normalizePassword(password);
+		const system = await this.systemRepo.findById(systemId);
+		const school = await this.schoolRepo.findById(schoolId);
+		if (!school.externalId) {
+			throw new UnauthorizedException();
+		}
+		const userNameWithSchool = `${school.externalId}/${username}`;
```

Comment:
It is a small logic line, but is is logic. If you want to seperate orchestration and logic clean it must be outsource to a private method, or on some other place.

Reply:
I don't see this method as an orchestrator of logic. It contains the logic to login a user. 
Only if the same logic is needed in other strategies I externalized it.

Reply:
```
private checkUserId(account) {
     if (!account.userId) {
		throw new UnauthorizedException();
	}
};
private async checkBrutForce(userNameWithSchool, systemId) {
        const account = await this.authenticationService.loadAccount(userNameWithSchool, systemId);
	this.checkUserId(account);
	this.authenticationService.checkBrutForce(account);
}